receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+// +build windows
+
+package k8swindows // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows"
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/hcsshim"
+	kubeletsummaryprovider "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type K8sWindows struct {
+	cancel                 context.CancelFunc
+	logger                 *zap.Logger
+	nodeName               string `toml:"node_name"`
+	k8sDecorator           stores.K8sDecorator
+	kubeletSummaryProvider *kubeletsummaryprovider.SummaryProvider
+	hcsshimStatsProvider   *hcsshim.HCSStatsProvider
+	hostInfo               host.Info
+	version                string
+}
+
+var metricsExtractors = []extractors.MetricExtractor{}
+
+func New(logger *zap.Logger, decorator *stores.K8sDecorator, hostInfo host.Info) (*K8sWindows, error) {
```

Comment:
There are no tests for this file methods

Reply:
I added tests for this file. Unit test coverage is only limited to initializing k8sWindows object. 
Adding unit test for GetMetrics will require some refactoring. Can follow up with adding unit test ater release.